### PR TITLE
add rdkit features

### PR DESCRIPTION
**Pull request by kjappelbaum**

### CODE REVIEW

File: data/tabular/rdkit_features/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+targets:
+    - id: formula
+      type: formula
+      names:
+          - noun: formula
+          - noun: chemical formula
+          - noun: chemical formula
+          - noun: molecular formula
+    - id: NumHDonors
+      type: text
```
Comment (MicPie) [inline data/tabular/rdkit_features/meta.yaml:14]:
```suggestion
      type: continuous
```

File: data/tabular/rdkit_features/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+      significant_digits: 1
+      names:
+          - noun: number of hydrogen bond donors
+          - noun: number of hydrogen bond donor sites
+          - noun: count of hydrogen bond donors
+    - id: NumHAcceptors
+      type: text
```
Comment (MicPie) [inline data/tabular/rdkit_features/meta.yaml:21]:
```suggestion
      type: continuous
```

File: data/tabular/rdkit_features/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+      significant_digits: 0
+      names:
+          - noun: number of hydrogen bond acceptors
+          - noun: number of hydrogen bond acceptor sites
+          - noun: count of hydrogen bond acceptors
+    - id: NumHeteroatoms
+      type: continuos
```
Comment (MicPie) [inline data/tabular/rdkit_features/meta.yaml:28]:
```suggestion
      type: continuous
```

File: data/tabular/rdkit_features/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+      significant_digits: 0
+      names:
+          - noun: number of heteroatoms
+          - noun: count of heteroatoms
+          - noun: heteroatom count
+    - id: RingCount
+      type: text
```
Comment (MicPie) [inline data/tabular/rdkit_features/meta.yaml:35]:
```suggestion
      type: continuous
```

File: data/tabular/rdkit_features/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+      significant_digits: 0
+      names:
+          - noun: number of rings
+          - noun: count of rings
+          - noun: ring count
+    - id: NumRotatableBonds
+      type: text
```
Comment (MicPie) [inline data/tabular/rdkit_features/meta.yaml:42]:
```suggestion
      type: continuous
```

File: data/tabular/rdkit_features/meta.yaml (hunk 2 above)
Comment (MicPie) [reply data/tabular/rdkit_features/meta.yaml:21]: or `categorical` for integers?

File: data/tabular/rdkit_features/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+      significant_digits: 0
+      names:
+          - noun: number of rotatable bonds
+          - noun: count of rotatable bonds
+          - noun: rotatable bond count
+    - id: NumAromaticBonds
+      type: text
```
Comment (MicPie) [inline data/tabular/rdkit_features/meta.yaml:49]:
```suggestion
      type: continuous
```

File: data/tabular/rdkit_features/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+      significant_digits: 0
+      names:
+          - noun: number of aromatic bonds
+          - noun: count of aromatic bonds
+          - noun: aromatic bond count
+    - id: NumAcidGroups
+      type: text
```
Comment (MicPie) [inline data/tabular/rdkit_features/meta.yaml:56]:
```suggestion
      type: continuous
```

File: data/tabular/rdkit_features/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+      significant_digits: 0
+      names:
+          - noun: number of acid groups
+          - noun: count of acid groups
+          - noun: acid group count
+    - id: NumBasicGroups
+      type: text
```
Comment (MicPie) [inline data/tabular/rdkit_features/meta.yaml:63]:
```suggestion
      type: continuous
```

File: data/tabular/rdkit_features/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+      significant_digits: 0
+      names:
+          - noun: number of basic groups
+          - noun: count of basic groups
+          - noun: basic group count
+    - id: Apol
+      type: text
```
Comment (MicPie) [inline data/tabular/rdkit_features/meta.yaml:70]:
```suggestion
      type: continuous
```

File: data/tabular/rdkit_features/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+      significant_digits: 2
+      names:
+          - noun: sum of atomic polarizabilities
+          - noun: total sum of atomic polarizabilities
+    - id: MolLogP
+      type: text
```
Comment (MicPie) [inline data/tabular/rdkit_features/meta.yaml:76]:
```suggestion
      type: continuous
```

**Review by MicPie — CHANGES_REQUESTED** [review]
Should be fine with the datatype changes. 👍

File: data/tabular/rdkit_features/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+      Answer: {NumAcidGroups#}.
+    - |-
+      Question: What is the {NumBasicGroups__names__noun} of the {#molecule|compound|chemical!} with {SMILES__description} {SMILES#}?
+      Answer: {NumBasicGroups#}.
+    - |-
+      Question: What is the {Apol__names__noun} of the {#molecule|compound|chemical!} with {SMILES__description} {SMILES#}?
+      Answer: {Apol#}.
+    - |-
+      User: I want to {#design|create|make|synthesize|analyze!} a {#molecule|compound|chemical!} with a {NumHDonors__names__noun} of {NumHDonors#} and a {NumHAcceptors__names__noun} of {NumHAcceptors#}.
+      Assistant: {#Cool, |Nice, |Interesting, |That's interesting, |That is a very interesting question, |!}Do you have some additional {#contraints|requirements|conditions|limitations|!}{# I should take into account| that help me narrow down the search| that I should consider| I should consider| I should take into account|!}?
```
Comment (MicPie) [inline data/tabular/rdkit_features/meta.yaml:144]:
```suggestion
      Assistant: {#Cool, do|Nice, do|Interesting, do|That's interesting, do|That is a very interesting question, do|Do!} you have some additional {#contraints|requirements|conditions|limitations|!}{# I should take into account| that help me narrow down the search| that I should consider| I should consider| I should take into account|!}?
```

Comment (MicPie) [reply data/tabular/rdkit_features/meta.yaml:144]: This needs to be fixed in other templates too.

File: data/tabular/rdkit_features/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+      Answer: {NumAromaticBonds#}.
+    - |-
+      Question: What is the {NumAcidGroups__names__noun} of the {#molecule|compound|chemical!} with {SMILES__description} {SMILES#}?
+      Answer: {NumAcidGroups#}.
+    - |-
+      Question: What is the {NumBasicGroups__names__noun} of the {#molecule|compound|chemical!} with {SMILES__description} {SMILES#}?
+      Answer: {NumBasicGroups#}.
+    - |-
+      Question: What is the {Apol__names__noun} of the {#molecule|compound|chemical!} with {SMILES__description} {SMILES#}?
+      Answer: {Apol#}.
```
Comment (MicPie) [inline data/tabular/rdkit_features/meta.yaml:141]:
```suggestion
      Answer: {Apol#}
```

Comment (MicPie) [reply data/tabular/rdkit_features/meta.yaml:141]: w/o`.`, or?

Comment (MicPie) [reply data/tabular/rdkit_features/meta.yaml:141]: this applies to other templates too

**Review by MicPie — CHANGES_REQUESTED** [review]

File: data/tabular/rdkit_features/meta.yaml (hunk 12 above)
Comment (kjappelbaum) [reply data/tabular/rdkit_features/meta.yaml:141]: i this case, I guess either is fine, but'll remove it

File: data/tabular/rdkit_features/meta.yaml (hunk 11 above)
Comment (kjappelbaum) [reply data/tabular/rdkit_features/meta.yaml:144]: wow, that is a great catch. I didn't even spot it when reading the generated text

File: data/text_sampling/text_sampling.py [hunk 13]
```diff
@@ -555,7 +555,7 @@ def _get_target_from_row(self, sample: pd.Series, var: str) -> str:
         ][0]
         data_type = var_dict["type"]
         if data_type == "continuous":
-            assert isinstance(out, float)
+            assert isinstance(out, (float, int)), f"out is not a float: {out}"
```
Comment (MicPie) [inline data/text_sampling/text_sampling.py:562]: Nitpicky: out is not a number (in or float)

File: data/tabular/rdkit_features/meta.yaml [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+targets:
+    - id: formula
+      type: formula
+      names:
+          - noun: formula
+          - noun: chemical formula
+          - noun: chemical formula
+          - noun: molecular formula
+    - id: NumHDonors
+      type: continuos
```
Comment (MicPie) [inline data/tabular/rdkit_features/meta.yaml:14]:
```suggestion
      type: continuous
```
this needs also replacement below

**Review by MicPie — CHANGES_REQUESTED** [review]
fix `continuous` ;-)

**Review by MicPie — APPROVED** [review]
lgtm

**Pull request merged**